ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/UpdateNormalizer.scala

```diff
@@ -29,6 +29,26 @@ object UpdateNormalizer {
   }
 }
 
+/** Normalizes the expected and/or actual updates based on their counterparts.
+  * Useful when scalar update normalization is not possible.
+  * (e.g. ensure contents of one update are a superset of the ones in the other update)
+  */
+trait PairwiseUpdateNormalizer {
+  def normalize(expectedUpdate: Update, actualUpdate: Update): (Update, Update)
+}
+
+object PairwiseUpdateNormalizer {
+  def normalize(
+      expectedUpdate: Update,
+      actualUpdate: Update,
+      normalizers: Iterable[PairwiseUpdateNormalizer],
+  ): (Update, Update) = {
+    normalizers.foldLeft(expectedUpdate -> actualUpdate) { case ((expected, actual), normalizer) =>
+      normalizer.normalize(expected, actual)
+    }
+  }
```

Comment:
You can remove the body's curly braces here.

Reply:
Done